public/src/components/campaign/campaign_list.jsx

```diff
@@ -14,16 +14,23 @@ export default class CampaignList extends Component {
 
   render() {
     const {
-      all_campaigns,
       account_info,
+      addCampaignToUser,
+      all_campaigns,
       auth,
       history,
-      addCampaignToUser
+      setCurrentCampaign
     } = this.props;
     const thisPage = 'Campaign';
     const { admin_campaigns, volunteer_campaigns } = buttons_obj;
     const { redirect_path } = campaignHeaders;
 
+    function handleClick(id, campaign_id, historyObj, campaign_list) {
```

Comment:
I think we were handling the click before in the button_objs component. Not sure if we want to remain consistent across all buttons?

Reply:
this is the function that gets passed into the button_objs component.